### PR TITLE
Changes

### PR DESCRIPTION
**Pull request by OkemmaduNancy**

### CODE REVIEW

File: src/APIs.js [hunk 1]
```diff
@@ -21,7 +21,6 @@ class API extends Component {
         }
     }
     render() {
-        // this.state.newData.map((item, index) => console.log(<h1 key={index}>{item.price}</h1>))
         return (
             <div>
                 { this.state.newData.map((item, index) => <h1 key={index}>{item.price}</h1>)}
```
Comment (EngrEric) [inline src/APIs.js:26]: new data is not verbose. Give your variable names with meaning for easy readability of codes

File: src/App.js [hunk 2]
```diff
@@ -1,61 +1,19 @@
...
-//   }
-// }
 
-// const obj = new Mydate()
-
-// const myName = obj.getMeOut()
 
 class App extends Component {
   state = {
     ProductData: ProductData,
```
Comment (EngrEric) [inline src/App.js:12]: variable names starts with small letter, i.e camel case

File: src/App.js [hunk 3]
```diff
@@ -66,27 +24,9 @@ class App extends Component {
...
-  //  }
-
   render() {
     return (
-      // <BrowserRouter>
-      // <Switch>
-      //   <Route path="/new" component={this.NewRoute} />
-
       <div className="App">
-        {/* <ButtonClick /> */}
```
Comment (EngrEric) [inline src/App.js:30]: why the space here

File: src/Component/AboutUs.js [hunk 4]
```diff
@@ -55,7 +55,7 @@ const AboutUs = () => {
         Pellentesque volutpat porttitor nunc, quis sollicitudin velit aliquet
         vitae.
       </p>
-      <button onClick={() => history.push("/contact-us")} className={classes.button}>Read More</button>
+      <button onClick={() => history.push("/App")} className={classes.readMoreButton}>Read More</button>
```
Comment (EngrEric) [inline src/Component/AboutUs.js:58]: route names should start with small letters and separated by hyphens

File: src/Component/Card.js [hunk 5]
```diff
@@ -32,12 +33,16 @@ const Card = ({ imageUrl, description, id, btn }) => {
...
-      <img src={imageUrl} className={classname} />
-      <button>{btn}</button>
+      <Link to={`/product/${id}`}>
+        <img className={classes.image} src={imageUrl} />
+        <div>{description}</div>
+      </Link>
+      <button>{card_button}</button>
       <Button type="primary" onClick={() => handledelete(id)}>delete</Button>
-      <p>{description}</p>
+
```
Comment (EngrEric) [inline src/Component/Card.js:42]: why space here

File: src/Component/Card.module.css [hunk 6]
```diff
@@ -1,42 +1,24 @@
...
+
+.image {
   padding-left: 30px;
   width: "200px";
   height: 200px;
 }
- button{
+/* p{
+  margin-left: 0px;
+} */
```
Comment (EngrEric) [inline src/Component/Card.module.css:20]: remove is you're not using it

File: src/Component/Home.js [hunk 7]
```diff
@@ -15,7 +14,7 @@ const Navbar = () => {
       </p>
       <p>
         <li>
-          <a href="contact-us">Contact Us</a>
+          <a href="sign-up">Sign Up</a>
```
Comment (EngrEric) [inline src/Component/Home.js:17]: `href="/sign-up"` notice the difference

File: src/Component/Nav/Burger.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import styled from 'styled-components';
+
+const StyledBurger = styled.div`
+    
+
+  
+`
```
Comment (EngrEric) [inline src/Component/Nav/Burger.js:8]: why the space here

File: src/Component/Nav/Burger.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+const Burger = () => {
+    // const [open, setOpen] = useState(false)
+
```
Comment (EngrEric) [inline src/Component/Nav/Burger.js:11]: remove

File: src/Component/Nav/Burger.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    return (
+        <>
+            <StyledBurger >
+                <div />
+                <div />
+                <div />
```
Comment (EngrEric) [inline src/Component/Nav/Burger.js:17]: why empty divs

File: src/Component/Nav/RightNav.js [hunk 11]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import styled from 'styled-components';
+import Checkbox from './Checkbox';
+
+const Nav = styled.nav`
+
```
Comment (EngrEric) [inline src/Component/Nav/RightNav.js:6]: space here, remove

File: src/Component/Nav/RightNav.js [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+    <Nav>
+      <Checkbox />
+
+      <nav >
+        <div className="right_header">
+          <h3 className="docs">DOCS</h3>
+          <h3 className="pres">Presentation</h3>
+          <button className="signIn" type="submit" >Sing In</button>
+          <button className="logIn" type="submit">Log In</button>
+
```
Comment (EngrEric) [inline src/Component/Nav/RightNav.js:60]: space here, remove

File: src/Component/ProductDetails.js [hunk 13]
```diff
@@ -1,23 +1,55 @@
-import React, { useState } from "react";
+import { message } from "antd";
+import { Content } from "antd/lib/layout/layout";
+import React, { useEffect } from "react";
+import { useHistory, useParams } from "react-router";
+import classes from './ProductDetails.module.css'
+const ProductDetails = () => {
```
Comment (EngrEric) [inline src/Component/ProductDetails.js:6]: add space btw import and component declaration

File: src/Component/ProductDetails.js [hunk 14]
```diff
@@ -1,23 +1,55 @@
...
 
-    // const { id } = props.match.params.id
+    const { imageUrl } = useParams()
 
-    async function productDetails() {
+
+    const [productDetails, setProductDetails] = React.useState({})
+    const [error, seterror] = React.useState("")
+
+
```
Comment (EngrEric) [inline src/Component/ProductDetails.js:16]: one space is okay here

File: src/Component/ProductDetails.js [hunk 15]
```diff
@@ -1,23 +1,55 @@
...
+    async function getProducts() {
         try {
             const details = await fetch(`http://localhost:4000/v1/product/${id}`, {
```
Comment (EngrEric) [inline src/Component/ProductDetails.js:19]: create a base use url  in constants.js and import here

//constants.js
`export BASR_URL = 'http://localhost:4000/v1'`

then import here and use as follows
` await fetch(`${BASR_URL }/product/${id}`,`

this way you  can use BASE_URL so many places. Also notice how i named my  constant string variable in upper case

File: src/Component/ProductDetails.js [hunk 16]
```diff
@@ -1,23 +1,55 @@
...
-
+            seterror(error.message)
+            message.error({ Content: "error occured" })
         }
     }
+
+    useEffect(() => {
+        getProducts()
+    }, [])
+    console.log(productDetails);
```
Comment (EngrEric) [inline src/Component/ProductDetails.js:35]: remove console after use. don't push for production

Comment (EngrEric) [inline src/Component/ProductDetails.js:35]: remove console after use. don't push for production

File: src/Component/ProductDetails.js [hunk 17]
```diff
@@ -1,23 +1,55 @@
...
+            </div>
+            <div className={classes.info}>
+                <div className={classes.name}  >{productDetails.name}</div>
+                <div className={classes.price}>{productDetails.price} </div>
+                <div className={classes.cart}>{productDetails.cart}</div>
+                <div className={classes.categories}>{productDetails.categories}</div>
+                <div className={classes.description}>{productDetails.description}</div>
+            </div>
+            <div>
+
```
Comment (EngrEric) [inline src/Component/ProductDetails.js:49]: remove space

File: src/Component/ProductDetails.module.css [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+    color: white;
+    padding: 10px;
+    margin-top: 10px;
+    text-align: center;
+    border-radius: 2px;
+    cursor: pointer;
+} 
+
+.categories{
+
```
Comment (EngrEric) [inline src/Component/ProductDetails.module.css:36]: remove space

File: src/Component/ProductDetails.module.css [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+    margin-bottom: 10px; 
+    padding: 10px;
+    border-radius: 2px;
+    text-align: center;
+    background-color: rgb(230, 230, 230);
+}
+
+
+
+
```
Comment (EngrEric) [inline src/Component/ProductDetails.module.css:56]: remove space

File: src/Component/SignUp.js [hunk 20]
```diff
@@ -0,0 +1,186 @@
...
+  const [lastname, setLastName] = useState("");
+  const [age, setAge] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [dateOfBirth, setDateOfBirth] = useState("");
+  const [uploadPhoto, setUploadPhoto] = useState("");
+  const [gender, setGender] = useState("");
+  const [maritalStatus, setMaritalStatus] = useState("");
+  const [country, setCountry] = useState("");
+  const [job, setJob] = useState("");
```
Comment (EngrEric) [inline src/Component/SignUp.js:26]: rewrite all these, 
`const [userInputs, setUserInputs] = useState({})`

File: src/Component/SignUp.js [hunk 21]
```diff
@@ -0,0 +1,186 @@
...
+      email: email,
+      age: age,
+      dateOfBirth: dateOfBirth,
+      uploadPhoto: uploadPhoto,
+      gender: gender,
+      maritalStatus: maritalStatus,
+      country: country,
+      job: job
+
+    }
```
Comment (EngrEric) [inline src/Component/SignUp.js:54]: remove

File: src/Component/SignUp.js [hunk 22]
```diff
@@ -0,0 +1,186 @@
...
+    console.log(user);
```
Comment (EngrEric) [inline src/Component/SignUp.js:55]: `console.log(userInput)` will give you all the details

File: src/Component/SignUp.js [hunk 23]
```diff
@@ -0,0 +1,186 @@
...
+  };
+  const handlePassword = e => {
+    setPassword(e.target.value)
+  };
+  const handleDateOfBirth = e => {
+    setDateOfBirth(e.target.value)
+  };
+  const handleUpload = e => {
+    setUploadPhoto(e.target.value)
+  };
```
Comment (EngrEric) [inline src/Component/SignUp.js:78]: remove all the handle this and that. Replace with one function for all the inputs except confirm password

```
const handleInputChange = (event) => {
const name = event.target.name
const value = event.target.value
setUserInput({ ...userInput, [name]: value})
```

File: src/Component/SignUp.js [hunk 24]
```diff
@@ -0,0 +1,186 @@
...
+    setJob(e.target.value)
+  };
+
+
+
+
+  // useEffect(() => {
+  //   passwordConfirm()
+  // }, [])
+
```
Comment (EngrEric) [inline src/Component/SignUp.js:98]: remove

File: src/Component/SignUp.js [hunk 25]
```diff
@@ -0,0 +1,186 @@
...
+        <div className={classes.signUp_form}>
+          Last Name:
+            <input onChange={handleLastName} type="text" value={lastname} />
+        </div>
+        <div className={classes.signUp_form}>
+          Age:
+            <input onChange={handleAge} type="number" value={age} />
+        </div>
+        <div className={classes.signUp_form}>
+          Email:
```
Comment (EngrEric) [inline src/Component/SignUp.js:119]: replae al handle functions for input to `handleInputChange `

File: src/Component/SignUp.module.css [hunk 26]
```diff
@@ -0,0 +1,43 @@
...
+    margin: 0 auto;
+    width: 650px;
+    margin-bottom: 50px;
+    display: grid;
+    justify-content: center; 
+    grid-template-columns: 200px 200px 200px; 
+    grid-template-rows: 100px 100px; 
+    box-shadow: 0 10px 20px rgba(0,0,0,0.19);    
+  
+
```
Comment (EngrEric) [inline src/Component/SignUp.module.css:15]: remove space

File: src/Component/SignUp.module.css [hunk 27]
```diff
@@ -0,0 +1,43 @@
...
+}
+.signUp_form{
+    margin: 50px;
+    margin-bottom: 50px;
+    justify-self: center;
+    grid-row-gap: 10px;
+
```
Comment (EngrEric) [inline src/Component/SignUp.module.css:22]: remove space

File: src/Component/SignUp.module.css [hunk 28]
```diff
@@ -0,0 +1,43 @@
...
+    text-align: center;
+}
+
+.signUp_button{
+    margin-bottom: 40px;
+    width: 200px;
+    position: relative;
+    left: 42%;
+}
+i{
```
Comment (EngrEric) [inline src/Component/SignUp.module.css:37]: give space

File: src/Component/Store.js [hunk 29]
```diff
@@ -1,15 +1,21 @@
 import React, { useEffect, useState } from "react";
 import classes from "./Store.module.css";
 import Card from "./Card";
+import Navbar from "./Nav/RightNav";
 import { data } from "./data";
+import Checkbox from "./Nav/Checkbox";
+import Burger from "./Nav/Burger";
+
+
+
```
Comment (EngrEric) [inline src/Component/Store.js:10]: space is much

File: src/Form/Form.js [hunk 30]
```diff
@@ -31,47 +35,36 @@ const Form = () => {
...
+                        Name:
           <input name="name" onChange={handleChange} />
                     </label>
-                    <br />
-                    <label>
+                    <label className={classes.label1}>
                         Quantity:
           <input name="quantity" onChange={handleChange} />
                     </label>
-                    <br />
```
Comment (EngrEric) [inline src/Form/Form.js:47]: not aligned

File: src/Form/Form.js [hunk 31]
```diff
@@ -31,47 +35,36 @@ const Form = () => {
...
+                    <label className={classes.label1}>
                         Categories:
           <input name="categories" onChange={handleChange} />
                     </label>
-                    <br />
                 </fieldset>
-                <button type="submit">Submit</button>
+                <button className={classes.form_button} onClick={() => history.push("/store")} type="submit">Submit</button>
             </form>
         </div>
```
Comment (EngrEric) [inline src/Form/Form.js:69]: align properly

File: src/Form/form.module.css [hunk 32]
```diff
@@ -0,0 +1,38 @@
...
+}
+/* @media only screen and (max-width: 600px)  {
+        .Form {
+          display: grid;
+          grid-template-rows:  100px 100px ;
+        }
+      
+    } */
+
+input:hover {
```
Comment (EngrEric) [inline src/Form/form.module.css:27]: remove all commented codes

File: src/Router.js [hunk 33]
```diff
@@ -1,24 +1,25 @@
...
-import ContactUs from "./Component/ContactUs";
-
+import SignUp from "./Component/SignUp";
+import ProductDetails from "./Component/ProductDetails";
 class Routes extends Component {
   render() {
     return (
       <Router>
         <Suspense fallback={<h1>loading</h1>}>
           <Switch>
```
Comment (EngrEric) [inline src/Router.js:15]: replace this `<h1>loading</h1>`to a loading component, maybe spinning one

**Review by EngrEric — CHANGES_REQUESTED** [review]
You need to make the changes from the comments

File: src/Component/Nav/Burger.js (hunk 10 above)
Comment (OkemmaduNancy) [reply src/Component/Nav/Burger.js:17]: for creating a hamburger

**Pull request merged**